llvm/docs/CIBestPractices.rst

```diff
@@ -0,0 +1,86 @@
+======================
+LLVM CI Best Practices
+======================
+
+Overview
+========
+
+This document contains a list of guidelines and best practices to use when
+working on LLVM's CI systems. These are intended to keep our actions reliable,
+consistent, and secure.
+
+Github Actions Best Practices
+=============================
+
+This section contains information on best practices/guidelines when working on
+LLVM's github actions workflows.
+
+Disabling Jobs In Forks
+-----------------------
+
+There are many LLVM forks that exist, and we currently default to preventing
+actions from running outside of the LLVM organization to prevent them from
+running in forks. We default to this as actions running in forks are usually
+not desired and only run by accident. In addition, many of our workflows
+assume that they are operating within the main LLVM repository and break
+otherwise.
+
+Adhering to this best practice looks like adding the following to each of the
+jobs specified within a workflow:
+
+.. code-block:: yaml
+
+  jobs:
+    <job name>:
+      if: github.repository_owner == 'llvm'
+
+We choose to use ``github.repository_owner`` rather than ``github.repository``
+to enable these workflows to run in forks inside the LLVM organization such as
+the ClangIR fork.
+
+Hash Pinning Dependencies
```

Comment:
It might be worth adding a note about revisiting this bit once GitHub has fully rolled out immutable actions.

Reply:
Thanks, added a note.